File: archives/2017-2018/lectures/13/Books/app/src/main/java/ro/ubbcluj/cs/books/db/BooksDao.java
package ro.ubbcluj.cs.books.db;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

import ro.ubbcluj.cs.books.domain.Book;

@Dao
public interface BooksDao {

  @Insert
  void addBook(Book book);

  @Insert
  void addBooks(List<Book> books);

  @Delete
  void deleteBook(Book b);

  @Query("delete from books")
  void deleteBooks();

  @Update
  void updateBook(Book b);

  @Query("select * from books")
  LiveData<List<Book>> getBooks();
}

File: archives/2018-2019/13/Books/app/src/main/java/ro/ubbcluj/cs/books/db/BooksDao.kt
package ro.ubbcluj.cs.books.db

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.Dao
import android.arch.persistence.room.Delete
import android.arch.persistence.room.Insert
import android.arch.persistence.room.Query
import android.arch.persistence.room.Update

import ro.ubbcluj.cs.books.domain.Book

@Dao
interface BooksDao {

  @get:Query("select * from books")
  val books: LiveData<MutableList<Book>>

  @Insert
  fun addBook(book: Book)

  @Insert
  fun addBooks(books: List<Book>)

  @Delete
  fun deleteBook(b: Book)

  @Query("delete from books")
  fun deleteBooks()

  @Update
  fun updateBook(b: Book)
}
